docs/lang/logic.md

```diff
@@ -0,0 +1,458 @@
+Apalache produces a static type error, if the type of `e_1` and `e_2` do not
+match.
+
+**Example in TLA+:**
+
+```tla
+  FALSE = FALSE         \* TRUE
+  FALSE = TRUE          \* FALSE
+  10 = 20               \* FALSE
+  15 = 15               \* TRUE
+  "Hello" = "world"     \* FALSE
+  "Hello" = "hello"     \* FALSE
+  "Bob" = "Bob"         \* TRUE
+  { 1, 2 } = { 2, 3}    \* FALSE
+  { 1, 2 } = { 2, 1}    \* TRUE
+  { 1 } \ { 1 } = { "a" } \ { "a" } \* TRUE in pure TLA+ and TLC,
+                                    \* type error in Apalache
+  { { 1, 2 } } = { { 1, 2, 2, 2 } } \* TRUE
+  <<1, "a">> = <<1, "a">>           \* TRUE
+  <<1, "a">> = <<1, "b">>           \* FALSE
+  <<1, FALSE>> = <<2>>              \* FALSE in pure TLA+ and TLC,
+                                    \* type error in Apalache
+  <<1, 2>> = <<1, 2, 3>>            \* FALSE in pure TLA+ and TLC,
+                                    \* FALSE in Apalache, when both values
+                                    \* are treated as sequences
+  [ a |-> 1, b |-> 3 ] = [ a |-> 1, b |-> 3 ]           \* TRUE
+  [ a |-> 1, b |-> 3 ] = [ a |-> 1 ]                    \* FALSE
+  [ x \in 2..2 |-> x + x ] = [ x \in {2} |-> 2 * x ]    \* TRUE
+  [ x \in 2..3 |-> x + x ] = [ x \in {2, 3} |-> 2 * x ] \* FALSE
+```
+
+**Example in Python:** The standard data structures also implement
+    structural equality in Python, though we have to be careful to
+    use `==` instead of `=`:
+
+```python
+  False == False
+  False == True
+  10 == 20
+  15 == 15
+  "Hello" == "world"
+  "Hello" == "hello"
+  "Bob" == "Bob"
+  frozenset({ 1, 2 }) == frozenset({ 2, 3 })
+  frozenset({ 1, 2 }) == frozenset({ 2, 1 })
+  frozenset({ 1 }) - frozenset({ 1 }) == frozenset({ "a" }) - frozenset({ "a" })
+  frozenset({ frozenset({ 1, 2 }) }) == frozenset({ frozenset({ 1, 2, 2, 2 }) })
+  (1, "a") == (1, "a")
+  (1, "a") == (1, "b")
+  (1, False) == (2, )
+  (1, 2) == (1, 2, 3)
+  { "a": 1, "b": 3 } == { "a": 1, "b": 3 }
+  { "a": 1, "b": 3 } == { "a": 1 }
+  { x: (x + x) for x in { 2 } } == { x: (x * x) for x in { 2 } }
+  { x: (x + x) for x in { 2, 3 } } == { x: 2 * x for x in { 2, 3 } } 
+```
+
+----------------------------------------------------------------------------
+
+<a name="neq"></a>
+### Inequality
+
+**Notation:** `e_1 /= e_2` or `e_1 # e_2`
+
+**LaTeX notation:** ![ne](./img/ne.png)
+
+**Arguments:** Two arguments.
+
+**Effect:** This operator is syntax sugar for `~(e_1 = e_2)`. Full stop.
+
+----------------------------------------------------------------------------
+
+<a name="chooseBounded"></a>
+### Bounded Choice
+
+_This operator causes a lot of confusion. Read carefully!_
+
+**Notation:** `CHOOSE x \in S: P`
+
+**LaTeX notation:** ![choose-bounded](./img/choose-bounded.png)
+
+**Arguments:** Three arguments: a variable name, a set, and an
+expression.
+
+**Effect:** This operator is a magical device by the definition, no kidding. It
```

Comment:
I don't understand this sentence.

Reply:
Agreed, tone down the theatrics in a manual.